components/empty-state/empty-state-illustrated-button.js

```diff
@@ -1,63 +1,139 @@
+import { html, LitElement, nothing } from 'lit';
+import { bodyCompactStyles } from '../typography/styles.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { loadSvg } from '../../generated/empty-state/presetIllustrationLoader.js';
+import ResizeObserver from 'resize-observer-polyfill/dist/ResizeObserver.es.js';
 import { RtlMixin } from '../../mixins/rtl-mixin.js';
 import { runAsync } from '../../directives/run-async/run-async.js';
+import { styleMap } from 'lit/directives/style-map.js';
+import { unsafeSVG } from 'lit/directives/unsafe-svg.js';
 
 /**
  * The `d2l-empty-state-illustrated-button` component is an empty state component that displays an illustration and action button. The illustration property can be set to use one of the preset illustrations or a custom SVG illustration can be added in the default slot.
  * @fires d2l-empty-state-action - Dispatched when the action button is clicked
- * @slot - Custom SVG content if illustration property is not set
+ * @slot - Custom SVG content if `illustration-name` property is not set
  */
 class EmptyStateIllustratedButton extends RtlMixin(LitElement) {
 
 	static get properties() {
 		return {
 			/**
 			 * Optional: The action text to be used in the subtle button
-			 * @type {String}
+			 * @type {string}
 			 */
 			actionText: { type: String, attribute: 'action-text' },
 			/**
 			 * REQUIRED: A description giving details about the empty state
-			 * @type {String}
+			 * @type {string}
 			 */
 			description: { type: String },
 			/**
 			 * Optional: The name of the preset image you would like to display in the component
-			 * @type {String}
+			 * @type {string}
 			 */
-			illustration: { type: String },
+			illustrationName: { type: String, attribute: 'illustration-name' },
+			/**
+			 * Optional: This will change the action button to use a primary button instead of the default subtle button
+			 * @type {boolean}
+			 */
+			primary: { type: Boolean },
 			/**
 			 * REQUIRED: A title for the empty state
-			 * @type {String}
+			 * @type {string}
 			 */
-			title: { type: String }
+			titleText: { type: String, attribute: 'title-text' },
+			_contentWidth: { type: Number }
 		};
 	}
 
 	static get styles() {
-		return [emptyStateStyles, emptyStateIllustratedStyles, bodyCompactStyles, heading2Styles];
+		return [bodyCompactStyles, emptyStateStyles, emptyStateIllustratedStyles];
+	}
+
+	constructor() {
+		super();
+
+		this._contentWidth = 700;
+		this._resizeObserver = new ResizeObserver(this._onResize.bind(this));
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this._resizeObserver.observe(this);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		this._resizeObserver.disconnect();
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		if (!this.titleText || this.titleText.length === 0) {
+			console.warn('d2l-empty-state-illustrated-button component requires titleText.');
+		}
+
```

Comment:
I don't think this is going to work in practice -- since often these are going to be lang terms that may not be set immediately on first-render.

Currently the only way we've successfully handled required attributes like this is to wait a few seconds after render (or update) and then check. Here's [a recent example in navigation](https://github.com/BrightspaceUILabs/navigation/blob/main/d2l-navigation-link-icon.js#L98). Longer term, we may want to look into some kind of mixin/helper/controller that could help make this kind of thing easier.

Reply:
I'm guessing there are cases where they're giving false negatives, yeah.